packages/react/src/ButtonGroup/ButtonGroup.test.tsx

```diff
@@ -0,0 +1,65 @@
+import React, {render, cleanup} from '@testing-library/react'
+import '@testing-library/jest-dom'
+
+import {ButtonGroup} from './ButtonGroup'
+import {Button} from '../Button'
+import {axe, toHaveNoViolations} from 'jest-axe'
+
+expect.extend(toHaveNoViolations)
+
+describe('ButtonGroup', () => {
+  afterEach(cleanup)
+
+  it('renders correctly into the document', () => {
+    const expectedClass = 'ButtonGroup'
+    const expectedTag = 'section'
+
+    const {getByTestId} = render(
+      <ButtonGroup data-testid="test">
+        <Button>Primary Action</Button>
+        <Button>Secondary Action</Button>
+      </ButtonGroup>
+    )
+    const buttonGroupEl = getByTestId('test')
+    expect(buttonGroupEl.tagName).toBe(expectedTag.toUpperCase())
+    expect(buttonGroupEl.classList).toContain(expectedClass)
+  })
+
+  it('renders buttons with the correct element type when buttonAs is set', () => {
+    const expectedTag = 'a'
+
+    const {getByTestId} = render(
+      <ButtonGroup buttonsAs={'a'}>
```

Comment:
Nit for consistency and unnecessary interpolation..

```suggestion
      <ButtonGroup buttonsAs="a">
```